From 3a68dc0b045962c3c72b0cd2af16a16d47ea86c2 Mon Sep 17 00:00:00 2001
From: Maksim Sisov <msisov@igalia.com>
Date: Tue, 13 Feb 2018 10:09:18 +0200
Subject: [PATCH] [ozone/wayland] Explicitly release capture of destroyed
 windows.

This unfreezes events when menu windows are closed with touch events.
---
 ui/ozone/platform/wayland/wayland_window.cc | 1 +
 1 file changed, 1 insertion(+)

diff --git a/ui/ozone/platform/wayland/wayland_window.cc b/ui/ozone/platform/wayland/wayland_window.cc
index 58d7b93266b8..1962f457fcf4 100644
--- a/ui/ozone/platform/wayland/wayland_window.cc
+++ b/ui/ozone/platform/wayland/wayland_window.cc
@@ -80,6 +80,7 @@ WaylandWindow::WaylandWindow(PlatformWindowDelegate* delegate,
       state_(PlatformWindowState::PLATFORM_WINDOW_STATE_UNKNOWN) {}
 
 WaylandWindow::~WaylandWindow() {
+  ReleaseCapture();
   PlatformEventSource::GetInstance()->RemovePlatformEventDispatcher(this);
   connection_->RemoveWindow(surface_.id());
   if (parent_window_)
-- 
2.11.0

